build_tools/cmake/riscv.toolchain.cmake

```diff
@@ -55,6 +56,30 @@ if(RISCV_CPU STREQUAL "linux-riscv_64")
     "--iree-llvmcpu-target-cpu-features=+m,+a,+f,+d,+c,+zvl512b,+v"
     "--riscv-v-fixed-length-vector-lmul-max=8"
     CACHE INTERNAL "Default llvm codegen flags for testing purposes")
+elseif(RISCV_CPU STREQUAL "generic-riscv_64")
```

Comment:
Looks like several flags are duplicated across linux/generic and 32/64 here. Maybe we'd want to refactor?

```cmake
if(RISCV_CPU MATCHES "generic")
  # disable OS features
elseif(RISCV_CPU MATCHES "linux")
  # enable OS features
endif()

if(RISCV_CPU MATCHES "riscv_32")
  # set 32-bit flags
elseif(RISCV_CPU MATCHES "riscv_64")
  # set 64-bit flags
endif()
```

Reply:
Ok, I'll refactor it.

Reply:
I can pull duplicate code out of the this codition. But some code can't pull out. for example:
`linux-riscv_32` has `RISCV32_TEST_DEFAULT_LLVM_FLAGS`.
`generic-riscv64` has additional `RISCV_COMPILER_FLAGS`

So we still need the `if(RISCV_CPU STREQUAL "generic-riscv_64")` condition.

Reply:
Good points. I think it could still be useful to restructure a bit, as that would make it more clear which options are shared based on a general property and which are unique to a specific configuration.